### ISSUE TITLE
Update to Mednafen 1.32.1 stable

### ISSUE THREAD

**Issue by PoloniumRain** [libretro/beetle-saturn-libretro#36]
Just a heads up - [Mednafen 1.32.1](https://forum.fobby.net/index.php?t=rview&goto=7029&th=2191#msg_7029) is now released and it's the first stable release in over 2 years. Obviously it would be great if Beetle Saturn was updated to 1.32.1.

**Comment by antoniodesousa** [comment]
Yes, please update the core to the latest stable release 🙏

**Comment by srk15372** [comment]
It would be marvelous hope it will happen regards

**Comment by Tasosgemah** [comment]
An update would be nice, there are plenty f Saturn fixes since the last time the core got updated.

**Comment by Tasosgemah** [comment]
An update would be nice, there are plenty of Saturn fixes since the last time the core got updated.

**Comment by beudbeud** [comment]
We want ST-V support please

**Comment by riggles1** [comment]
> We want ST-V support please

https://github.com/libretro/beetle-saturn-libretro/pull/45

Seems like we might be in luck and get it next year possibly :) 
Time is the issue right now for [vanfanel](https://github.com/vanfanel)

**Comment by beudbeud** [comment]
No new about it?

**Comment by Kdash4** [comment]
Still not updated? ☹️

**Comment by Vamp898** [comment]
+1 wishing for an update =) currently using mednafen directly as a workaround as it has some bugfixes that affected me but that is only an solution for the PC. Hoping to be able to use it in RetroArch =)

Also this should close the two open Grandia Bugs

**Comment by antoniodesousa** [comment]
It has been 1 year and 6 months since the request was made.
It probably never gonna happen like a bunch of other cores.

**Comment by Tasosgemah** [comment]
Wonder why there are no updates for this core anymore. Is it because its hard to implement them or because there is no dev who is interested?

**Comment by antoniodesousa** [comment]
I guess both. It's not easy to update the core to the latest release and no devs are interested in doing it.

So we gonna be stuck with 1.31.0 forever... which is 3 years old by now.

**Comment by Vamp898** [comment]
The relationship between Mednafen and RetroArch is complicated because, to a certain extend, they have different design goals and/or opinions about proper implementation.

That is the reason why the Cores are not called mednafen but beetle because, unlike Genesis Plus GX and some other cores, they are not just the core but an fork of it.

They changed/modified the source code and the issue is, everytime mednafen releases a new upstream release, the RetroArch devs have to update their fork with their changes to this version.

And this is much more complex than just building a new core, they have to touch the source and modify it.

If that is a good thing or not is up to debate. As far as i know, the mednafen developers demanded the name change because they think, RetroArch made them worse and they don't wanted to be associated with it and if you look at the bugs here, you will often read "It works with Mednafen but not with the Beetle core in RetroArch".

This is due to complex reasons including that Mednafen wanted to be as accurate and close to original hardware as possible, but the RetroArch developers wanted quality of live improvements that the Mednafen developers were not willing to implement these.

So the RetroArch devs forked Mednafen and implemented them themselves which... yeah. Makes things complicated.

My personal two cents, i want to be as close to the original hardware, i don't need quality of live improvements, i just want to play the games as they were supposed to be. So i would be fine if they just port mednafen without modifications.

**Comment by Vamp898** [comment]
I found this post in the Mednafen Forum
https://forum.fobby.net/index.php?t=msg&th=2231&start=0&

I assume this means, it will still take some time

**Comment by beudbeud** [comment]
There are this repository too 
https://github.com/libretro-mirrors/mednafen-git

**Comment by Tasosgemah** [comment]
> There are this repository too https://github.com/libretro-mirrors/mednafen-git

Is there a compiled dll we can try?

It seems older than this one (last updated 2y ago while this one is 7 months) but it's based on a more recent version.

**Comment by Vamp898** [comment]
First of all, i had to shrug when i read DLL, second, that is an untouched mirror of the Mednafen repository.

There is a compiled *shrug* DLL *shrug* on the Mednafen Homepage: https://mednafen.github.io/releases/

Mednafen is _not_ available as an libretro Core, the RetroArch people have their own repository where they wrap/port Mednafen for libretro under the Name Beetle.

That is not, what is in the mentioned repository. This repository is simply for comparison sake of what changed inside Mednafen, unrelated to libretro.

If you want best Saturn/PS1 Support, you have to use Mednafen or accept the outdated libretro core. Currently there is no other option. I personally am using Mednafen on the PC due to this, i only use the libretro Core on my Smartphone/Tablet

**Comment by riggles1** [comment]
I use the Beetle core because I need native CRT switchres support, RA is perfect for actual CRT usage.
RetroArch has different reach and a lot more features no matter how you slice it. Also better suited for a variety of portable devices.

So hopefully this core does get the updates in the future.

**Comment by beudbeud** [comment]
Yes  I use Retroarch for can use my CRT and arcade cabinet, with upgrade to 1.32.1 will can be introduce Sega Titan (Sega STV)

**Comment by Tasosgemah** [comment]
Mednafen standalone also doesn't support CHDs but the RetroArch core does, which is a big deal.

> Yes I use Retroarch for can use my CRT and arcade cabinet, with upgrade to 1.32.1 will can be introduce Sega Titan (Sega STV)

You can use the MAME core for that.

**Comment by Vamp898** [comment]
> I use the Beetle core because I need native CRT switchres support, RA is perfect for actual CRT usage. RetroArch has different reach and a lot more features no matter how you slice it. Also better suited for a variety of portable devices.
> 
> So hopefully this core does get the updates in the future.

I assume you mean automatic switching support, because Mednafen does have native support for CRT Resolution/Refresh Rate by setting the right mode. But you're right that it doesn't have any type of automatic detection/switching support, you have to set the correct mode yourself. 

It can be automated via an script (which is easy as Mednafen is an CLI tool, it's a dream for doing things automated), but it does not exist out of the box.

Also Mednafen does not run on Android/macOS, it still runs on every portable Linux/Windows system but again, yes, it's not the same as RetroArch. (Most Android/iOS devices aren't great enough to run Mednafen at native speed to begin with)

Mednafen is focused on highest accuracy while RetroArch is more focused on comfort features, that is why there is almost no cooperation between these two projects to begin with so yes, they are for a different target group. 

But both are OpenSource projects, both need volunteers to work on it. Currently nobody had time to port it yet

On a side note, I would not use RetroArch for STV on a Arcade Cabinet, because usually you want highest accuracy in an fixed environment and don't need the comfort features of RetroArch. Especially input configuration for exotic hardware devices is much easier on Mednafen as you have native input controls. On Saturn Emulation, you calibrate an Saturn controller (A, B, C, X, Y, T, L, R, Start) and not an Xbox controller (B/A/Y/X and so on) that you then have to remap.

I have so many input issues with RetroArch on mobile and Desktops with remapping (Not just me, the RetroArch bug tracker is full with them), on an Arcade Cabinet, you absolutely want an "Just Works" solution which Mednafen.

MAME has lower Accuracy and higher input lag btw, so MAME is only an option if the game is unsupported by Mednafen. If it is supported, you get the better experience with Mednafen

**Comment by LibretroAdmin** [comment]
Should be updated now
